projects/sandbox/src/app/wizard/wizard.component.ts

```diff
@@ -0,0 +1,169 @@
+import { Component, Input } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { HttpErrorResponse } from '@angular/common/http';
+import { merge } from 'rxjs';
+import { DataPollingComponent } from '@shared';
+import { DatasetInfo, SubmissionResponseData, WizardField, WizardStep } from '../_models';
+import { SandboxService } from '../_services';
+import { ProtocolType } from '@shared';
+
+export interface ValidatorArrayHash {
+  [key: string]: Array<Validators>;
+}
+
+@Component({
+  selector: 'sb-wizard',
+  templateUrl: './wizard.component.html',
+  styleUrls: ['./wizard.component.scss']
+})
+export class WizardComponent extends DataPollingComponent {
```

Comment:
This is a general remark about this wizard. The question I'm having is whether this is meant to be a generic wizard component (i.e. that can be used for other wizards as well) or whether this wizard is really only meant for this particular instance.

In the first case, there are serious issues with generalization and abstraction. So I'm assuming the second case.
But then we should rename this component as well as all config classes and so on to reflect the specific use of this wizard. So this is a dataset submission wizard? Or a dataset tracking wizard? 

In any case, it seems this wizard is used both for submitting a new dataset as for tracking an already submitted one, is that right? (And is that why the step is initialized to show the last page of the wizard?) Seems a bit forced to me, and perhaps we should separate the dataset processing monitoring (which has to be available separately) away from this wizard?

Reply:
The functionality is configurable but not properly generic.  The progress display has its own component, and can therefore be dropped into any template, though it is true that its form is embedded within the wizard.

The progress display is not properly a "step" of data submission but having it behave as a pseudo-step it allows it to:
* work with the navigation arrows
* display when its orb is clicked
* use the `stepIsComplete` logic

It pulls the 2 forms together into one process - hopefully the sum of the changes made will make this seem less forced.